pyccl/cells.py

```diff
@@ -69,17 +103,52 @@ def angular_cl(cosmo, tracer1, tracer2, ell, *,
     if not (np.diff(ell_use) > 0).all():
         raise ValueError("ell values must be monotonically increasing")
 
+    if auto_limber or (type(l_limber) != str and ell_use[0] < l_limber):
```

Comment:
I don't think you need to check for the type of `l_limber` at this point. If it's not a number it will just fail.